src/redux/modules/courseSelect.js

```diff
@@ -1,10 +1,33 @@
 const TOGGLE_COURSE_SELECT = "bobcat-search/courseselect/TOGGLE_COURSE_SELECT";
-
+const CLEAR_SCHEDULE = "bobcat-search/courseselect/CLEAR_SCHEDULE";
 const initialState = {};
 
 export default function reducer(state = initialState, action = {}) {
   switch (action.type) {
     case TOGGLE_COURSE_SELECT:
+      //handle conflicting course
+      if (action.payload.conflicts !== undefined) {
+        return {
+          ...state,
+          [action.payload.semester + action.payload.year]: state[
+            action.payload.semester + action.payload.year
+          ].filter(function (course) {
```

Comment:
Let's use an arrow function here

Reply:
I'm not sure why. But whenever I convert to arrow function for this, redux would be called infinitely.

Reply:
Actually on a related note since we are returning a boolean, we can return a conditional expression. To make it more readable we could store each individual condition into a named variable that represents its value. Then we can return `!var1 && !var2` where var1 and var2 are the named variables.

Reply:
I saw my mistake now. It's now an arrow function